lib/services/platform-environment-requirements.ts

```diff
@@ -0,0 +1,135 @@
+			}
+
+			this.$logger.info("You are missing the nativescript-cloud extension and you will not be able to execute cloud builds. Your environment is not configured properly and you will not be able to execute local builds. " + EOL
+				+ "Select “Configure for Cloud Builds” to install the nativescript-cloud extension and automatically configure your environment for cloud builds." + EOL
+				+ "Select “Configure for Local Builds” to run the setup script and automatically configure your environment for local builds."
+				+ "Select “Configure for Both Local and Cloud Builds” to automatically configure your environment for both options."
+				+ "Select “Skip Step and Configure Manually” to disregard these options and install any required components manually.");
+
+			const selectedOption = await this.$prompter.promptForChoice(PlatformEnvironmentRequirements.NOT_CONFIGURED_ENV_MESSAGE, [
+				PlatformEnvironmentRequirements.CLOUD_BUILDS_OPTION_NAME,
+				PlatformEnvironmentRequirements.SETUP_SCRIPT_OPTION_NAME,
+				PlatformEnvironmentRequirements.MANUALLY_SETUP_OPTION_NAME,
+				PlatformEnvironmentRequirements.BOTH_CLOUD_BUILDS_AND_SETUP_SCRIPT_OPTION_NAME
+			]);
+
+			if (selectedOption === PlatformEnvironmentRequirements.CLOUD_BUILDS_OPTION_NAME) {
+				await this.processCloudBuilds(platform);
+			}
+
+			if (selectedOption === PlatformEnvironmentRequirements.SETUP_SCRIPT_OPTION_NAME) {
+				await this.$doctorService.runSetupScript();
+
+				if (await this.$doctorService.canExecuteLocalBuild(platform)) {
+					return true;
+				}
+
+				const option = await this.$prompter.promptForChoice(PlatformEnvironmentRequirements.NOT_CONFIGURED_ENV_AFTER_SETUP_SCRIPT_MESSAGE, [
+					PlatformEnvironmentRequirements.CLOUD_BUILDS_OPTION_NAME,
+					PlatformEnvironmentRequirements.MANUALLY_SETUP_OPTION_NAME
+				]);
+
+				if (option === PlatformEnvironmentRequirements.CLOUD_BUILDS_OPTION_NAME) {
+					await this.processCloudBuilds(platform);
+				}
+
+				if (option === PlatformEnvironmentRequirements.MANUALLY_SETUP_OPTION_NAME) {
+					this.processManuallySetup(platform);
+				}
+			}
+
+			if (selectedOption === PlatformEnvironmentRequirements.MANUALLY_SETUP_OPTION_NAME) {
+				this.processManuallySetup(platform);
+			}
+
+			if (selectedOption === PlatformEnvironmentRequirements.BOTH_CLOUD_BUILDS_AND_SETUP_SCRIPT_OPTION_NAME) {
+				await this.processBothCloudBuildsAndSetupScript(platform);
+				if (await this.$doctorService.canExecuteLocalBuild(platform)) {
+					return true;
+				}
+
+				this.processManuallySetup(platform);
+			}
+		}
+
+		return true;
+	}
+
+	private async processCloudBuilds(platform: string): Promise<void> {
+		await this.processCloudBuildsCore(platform);
+		this.fail(this.getCloudBuildsMessage(platform));
+	}
+
+	private async processCloudBuildsCore(platform: string): Promise<IExtensionData> {
+		const installedExtensions = this.$extensibilityService.getInstalledExtensions();
```

Comment:
This whole logic - "check if ns-cloud is installed and if it's not install it" could be extracted to a small service and reused here and in the `cloud setup` command